docs/basics/101-113-symlinks.rst

```diff
@@ -0,0 +1,222 @@
+   subsequent sections of the book. But it can help to establish trust in that
+   your data is safely stored and tracked, and it can get certainly helpful
+   should you be one of those weird people that always want to understand
+   things in depth (those people are great, btw!). Also, certain file management operations
+   can be messy -- for example, you shouldn't attempt to move a subdirectory
+   (more on this in a dedicated section <link>), because it can break symlinks.
+   Understanding more about the object tree can help to understand such
+   problems, and knowing bits of the Git-annex basics can make you more
+   confident in working with your datasets.
+
+   So how do these paths and names come into existence?
+
+   When a file is annexed, Git-annex generates a *key* from the file content
+   that it uses (in part) as a name for the file and as a part of the path
+   in the object tree.
+   The key is associated with the content of the file (the *value*).
+   Therefore, using this key, file content can be (somewhat) identified
+   (or rather: Based on the keys, it can be identified whether two files
+   have identical contents, or whether file content changed).
+
+   The key is generated using *hashes*. A hash is a function that turns an
+   input (e.g. a PDF file) into a string of characters with a fixed length.
+   In principle, therefore, the hash function simply transforms a content of
+   any size into a string with fixed length.
+
+   The important aspect of a hash function is that it
+   will generate the same hash for the same file content, but once file content
+   changes, the generated hash will also look differently. If two files are
+   turned into identical character strings, the content in these files is thus
+   identical.
+
+   This key (or :term:`checksum`) is the last part of the name of the file the
+   symlink links to (in which the actual data content
+   is stored). The extension (e.g. ``.pdf``) is appended because some
+   operating systems need this information. The key is also one of the subdirectory
+   names in the path. This two-level structure is implemented because it helps to prevent
+   accidental deletions and changes.
+
+   .. runrecord:: _examples/DL-101-113-104
+      :language: console
+      :workdir: dl-101/DataLad-101/books
+
+      # take a look at the last part of the target path:
+      $ ls -lah TLCL.pdf
+
+   .. runrecord:: _examples/DL-101-113-105
+      :language: console
+      :workdir: dl-101/DataLad-101/books
+
+      # compare it to the checksum (here of type md5sum) of the PDF file and the subdirectory name
+      $ md5sum TLCL.pdf
+
+   There are different hash functions available. Depending on which is used,
+   the resulting :term:`checksum` has a certain length and structure.
+   By default, DataLad uses ``MD5e`` checksums, but should you want to, you can
+   change this default to `one of many other types <https://git-annex.branchable.com/backends/>`_.
```

Comment:
MD5 is used because the total path length of annex objects can become an issue https://github.com/datalad/datalad/issues/2571#issuecomment-404922110

Even though linux users don't really have to care, a dataset still has to consider the question whether it ever wants to be successully shared with a windows user....